Automattic/wp-calypso
# Plugins: add HOC to handle plugin redirect

#### Changes proposed in this Pull Request

This PR is a different approach to handle when the client should redirect once some plugins are installed on the site.
Almost all logic is handled by the `withPluginRedirect()` HOC, which wraps the `<PluginMeta />` component.
It supports the `WooCommerce` plugin redirection so far, but it gets almost everything ready to extend its functionality to scale to other plugins that require dealing with similar behavior.

Fixes #51327 #17686 https://github.com/Automattic/wp-calypso/issues/52539

#### Testing instructions

Test with **Jetpack** and **Atomic** site

* Go to WooCommerce plugin page in calypso: `<hostname>/plugins/woocommerce/<site-id>`

<img width="800" alt="Screen Shot 2021-04-26 at 9 30 50 AM" src="https://user-images.githubusercontent.com/77539/116083029-8606b080-a672-11eb-987e-0a6edc481c81.png">

* Click on the Install button to trigger the install process.
* Once the plugin is installed, the app should show the first success notice:
<img src="https://user-images.githubusercontent.com/77539/116469762-ecddc280-a848-11eb-83c6-c1f8dc2aec26.png" width="600px" />

* then, seconds after it should show another success message

<img src="https://user-images.githubusercontent.com/77539/116469529-a8522700-a848-11eb-8d55-48a2e49327c3.png" width="600px" />

* From here, confirm you have 5 seconds to cancel the redirect by clicking on the `Cancel` button.
* Confirm that after fice seconds, the client performs a redirect to the WooCommerce Wizard page.

<img width="800" alt="Screen Shot 2021-04-26 at 9 34 57 AM" src="https://user-images.githubusercontent.com/77539/116083166-b3ebf500-a672-11eb-9765-3854c621daab.png">

## Videos

### Jetpack site

https://user-images.githubusercontent.com/77539/116469530-a8522700-a848-11eb-8f59-30083ee73c22.mov

### Atomic site
https://user-images.githubusercontent.com/77539/116469456-91133980-a848-11eb-9381-5c2f2a4bbed9.mov





*



Related to #
